airbyte-integrations/connectors/source-shopify/source_shopify/spec.json

```
@@ -0,0 +1,25 @@
+{
+  "documentationUrl": "https://docs.airbyte.io/integrations/sources/shopify",
+  "connectionSpecification": {
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "Shopify Source CDK Specifications",
+    "type": "object",
+    "required": ["shop", "start_date", "api_password"],
+    "additionalProperties": false,
+    "properties": {
+      "shop": {
+        "type": "string",
+        "description": "The name of the shopify store. For https://EXAMPLE.myshopify.com, the shop name is 'EXAMPLE'."
+      },
+      "start_date": {
+        "type": "string",
+        "description": "Format: YYYY-MM-DD, from what date you would like to replicate the data?"
```

adding a pattern to the spec allows validating it in the UI (and soon in the connector itself)
```suggestion
        "description": "The date you would like to replicate data. Format: YYYY-MM-DD.",
        "examples": ["2021-01-24"]
        "pattern": "^[0-9]{4}-[0-9]{2}-[0-9]{2}$",
```

---

Didn't know about the pattern usage, thank you for the suggestion, edited it as well.